pdsfile/pds3file/__init__.py

```diff
@@ -0,0 +1,171 @@
+
+    # Logger
+    LOGGER = pdslogger.NullLogger()
+
+    # CACHE
+    DICTIONARY_CACHE_LIMIT = 200000
+    CACHE = pdscache.DictionaryCache(lifetime=cache_lifetime_for_class,
+                                     limit=DICTIONARY_CACHE_LIMIT,
+                                     logger=LOGGER)
+
+    LOCAL_PRELOADED = []
+    SUBCLASSES = {}
+
+    # Override the rules
+    DESCRIPTION_AND_ICON = rules.DESCRIPTION_AND_ICON
+    ASSOCIATIONS = rules.ASSOCIATIONS
+    VERSIONS = rules.VERSIONS
+    INFO_FILE_BASENAMES = rules.INFO_FILE_BASENAMES
+    NEIGHBORS = rules.NEIGHBORS
+    SIBLINGS = rules.SIBLINGS       # just used by Viewmaster right now
+    SORT_KEY = rules.SORT_KEY
+    SPLIT_RULES = rules.SPLIT_RULES
+    VIEW_OPTIONS = rules.VIEW_OPTIONS
+    VIEWABLES = rules.VIEWABLES
+    LID_AFTER_DSID = rules.LID_AFTER_DSID
+    DATA_SET_ID = rules.DATA_SET_ID
+
+    OPUS_TYPE = rules.OPUS_TYPE
+    OPUS_FORMAT = rules.OPUS_FORMAT
+    OPUS_PRODUCTS = rules.OPUS_PRODUCTS
+    OPUS_ID = rules.OPUS_ID
+    OPUS_ID_TO_PRIMARY_LOGICAL_PATH = rules.OPUS_ID_TO_PRIMARY_LOGICAL_PATH
+
+    OPUS_ID_TO_SUBCLASS = rules.OPUS_ID_TO_SUBCLASS
+    FILESPEC_TO_VOLSET = rules.FILESPEC_TO_VOLSET
+    FILESPEC_TO_BUNDLESET = FILESPEC_TO_VOLSET
+
+    def __init__(self):
+        super().__init__()
+        # alias for attributes
+        self.volset_  = self.bundleset_
+        self.volset   = self.bundleset
+        self.volname_ = self.bundlename_
+        self.volname  = self.bundlename
+
+    # Alias, compatible with old function/property names
+    @property
+    def is_volset(self):
+        return self.is_bundleset
+
+    @property
+    def is_volset_file(self):
+        return self.is_bundleset_file
+
+    @property
+    def is_volset_dir(self):
+        return self.is_bundleset_dir
+
+    @property
+    def is_volume(self):
+        return self.is_bundle
+
+    @property
+    def is_volume_file(self):
+        return self.is_bundle_file
+
+    @property
+    def is_volume_dir(self):
+        return self.is_bundle_dir
+
+    def log_path_for_volset(self, suffix='', task='', dir='', place='default'):
+        return self.log_path_for_bundleset(suffix, task, dir, place)
+
+    def log_path_for_volume(self, suffix='', task='', dir='', place='default'):
+        return self.log_path_for_bundle(suffix, task, dir, place)
+
+    # Override functions
+    def __repr__(self):
+        if self.abspath is None:
+            return 'Pds3File-logical("' + self.logical_path + '")'
+        elif type(self) == Pds3File:
+            return 'Pds3File("' + self.abspath + '")'
+        else:
+            return ('Pds3File.' + type(self).__name__ + '("' +
+                    self.abspath + '")')
+
+    ######################################################################################
+    # PdsLogger support
+    ######################################################################################
+    @classmethod
+    def set_logger(cls, self, logger=None):
```

Comment:
I approved too soon...`set_logger` and `set_easylogger` in both `Pds3File` and `Pds4File` need to have the `self` argument removed.

Reply:
Fixed